ouroboros-consensus-cardano/tools/db-analyser/Main.hs

```diff
@@ -0,0 +1,330 @@
+
+showBlockTxsSize
+  :: forall blk. (HasHeader blk, HasAnalysis blk, ImmDbSerialiseConstraints blk)
+  => Analysis blk
+showBlockTxsSize immDB rr = processAll immDB rr process
+  where
+    process :: blk -> IO ()
+    process blk = putStrLn $ intercalate "\t" [
+          show slotNo
+        , "Num txs in block = " <> show numBlockTxs
+        , "Total size of txs in block = " <> show blockTxsSize
+        ]
+      where
+        txs :: [BL.ByteString]
+        txs = Analysis.blockTxs blk
+
+        numBlockTxs :: Int
+        numBlockTxs = length txs
+
+        blockTxsSize :: Int64
+        blockTxsSize = foldl' (+) 0 $
+          map BL.length txs
+
+        slotNo = blockSlot blk
+
+{-------------------------------------------------------------------------------
+  Auxiliary: processing all blocks in the imm DB
+-------------------------------------------------------------------------------}
+
+processAll :: forall blk. (HasHeader blk, ImmDbSerialiseConstraints blk)
+           => ImmDB IO blk
+           -> ResourceRegistry IO
+           -> (blk -> IO ())
+           -> IO ()
+processAll immDB rr callback = do
+    tipPoint <- getPointAtTip immDB
+    case pointToWithOriginRealPoint tipPoint of
+      Origin -> return ()
+      At tip -> do
+        Right itr <- stream immDB rr GetBlock
+          (StreamFromExclusive genesisPoint)
+          (StreamToInclusive tip)
+        go itr
+  where
+    go :: Iterator (HeaderHash blk) IO (IO blk) -> IO ()
+    go itr = do
+        itrResult <- ImmDB.iteratorNext itr
+        case itrResult of
+          IteratorExhausted   -> return ()
+          IteratorResult mblk -> mblk >>= \blk -> callback blk >> go itr
+
+{-------------------------------------------------------------------------------
+  Command line args
+-------------------------------------------------------------------------------}
+
+data CmdLine = CmdLine {
+      clConfig    :: [FilePath]
+    , clIsMainNet :: Bool
+    , clImmDB     :: FilePath
+    , clBlockType :: Maybe BlockType
+    , clAnalysis  :: AnalysisName
+    } deriving Show
+
+data BlockType = Byron | Shelley | Cardano
+  deriving Show
+
+parseCmdLine :: Parser CmdLine
+parseCmdLine = CmdLine
+    <$> many (strOption (mconcat [
+            long "config"
+          , help "Path to config file or files. Multiple occurences of the\
+                 \option can be used"
+          , metavar "PATH"
+          ]))
+    <*> flag True False (mconcat [
+            long "testnet"
+          , help "The DB contains blocks from testnet rather than mainnet"
+          ])
+    <*> strOption (mconcat [
+            long "db"
+          , help "Path to the chain DB (parent of \"immutable\" directory)"
+          , metavar "PATH"
+          ])
+    <*> parseBlockType
+    <*> parseAnalysis
+
+parseBlockType :: Parser (Maybe BlockType)
+parseBlockType = asum [
+      flag' (Just Byron) $ mconcat [
+          long "byron"
+        , help "A network with ByronBlock's only"
```

Comment:
```suggestion
        , help "A network with ByronBlocks only"
```
In fact, I would go for "A Byron network". Similarly for the other two, i.e., "A Shelley network" and "A Byron-to-Shelley network".